changesLog.js

```diff
@@ -1,34 +1,75 @@
+}
+
 commander
   .version('1.0.0', '-v, --version')
   .usage('[OPTIONS]...')
-  .requiredOption('-m, --milestone <name>', 'must filter issue from milestone')
-  .option('-d, --duplicate <allow>', 'Should allow duplicated issues (Default: true)')
-  .option('-f, --filters <items>', 'comma separeted labels to filter issues', commaSeparatedList)
-  .option('-g, --groups <items>', 'comma separated labels to group by issues', commaSeparatedList)
+  .requiredOption('-m, --milestone <number>', 'must filter issue from milestone')
+  .option('-c, --customer <customer>, will filter the issues related to a customer if this option is received')
+  .option('-i, --includeAll <flag>', 'will include a section with issues done for all clients (Default: false)')
+  .option('-s, --state <state>', 'should use issues with state: closed, open, all (Default: closed)', checkValidStates)
+  .option('-f, --filters <labels>', 'comma separated labels to filter and group by issues', commaSeparatedList)
+  .option('-d, --duplicate <allow>', 'should allow duplicated issues in different groups (Default: true)')
   .parse(process.argv);
 
+let groupedIssues = [];
+const filters = commander.filters;
 const params = {
-  duplicate: commander.duplicate ? false : true,
   milestone: commander.milestone,
-  filters: commander.filters,
-  groups: commander.groups
+  state: commander.state ? commander.state : 'closed',
+  customer: commander.customer ? commander.customer : undefined,
+  duplicateIssues: commander.duplicate ? commander.duplicate : true,
+  includeIssueForAll: commander.includeAll ? commander.includeAll : true
 };
 
-console.log(`Generating Changelog for Milestone ${params.milestone}...`)
-console.log(`Duplicate issues (in different groups) set to: ${params.duplicate}.`)
-if (params.filters !== undefined) console.log(`Filters: ${params.filters}`);
-if (params.groups !== undefined) console.log(`Groups: ${params.groups}`);
+async function fetchIssues() {
+  console.log(`\n\nFetching issues in Milestone ${params.milestone}...\n`);
+  if(filters) {
+    groupedIssues = await fetchIssuesInMilestone(octokit, params, filters);
+    
+    if (groupedIssues.length > 0) {
+      console.log(`\nFinished fetching issues by filters in Milestone ${params.milestone}!\n`);
+      console.log(`Duplicate issues (in different groups) set to: ${params.duplicateIssues}.\n`);
+      console.log(`Started creating Changes log for issues based on Groups: ${filters}...\n`);
+      return true;
+    } else return false;
+  }
+  else {
+    groupedIssues = await fetchIssuesInMilestone(octokit, params);
+    
+    if (groupedIssues.length > 0) {
+      console.log(`\nFinished fecthing issues in Milestone ${params.milestone}!\n`);
+      console.log('Started creating Changes lof for issues in single block...\n');
+      return true;
+    } else return false;
+  }
+}
 
-printIssuesFromMilestone()
+async function asyncGenerateChangesLog() {
+  if(!octokit.asyncTryToken()) {
```

Comment:
I think it's a good pattern to avoid evaluating methods directly in conditionals? Mainly a style thing, but IMO also makes debugging easier. 

```suggestion
  const await isValidToken = octokit.asyncTryToken();
  if (!isValidToken) {
```